include/dlr_internal.h

```diff
@@ -0,0 +1,123 @@
+#ifndef DLR_INTERNAL_H_
+#define DLR_INTERNAL_H_
+
+#include <graph/graph_runtime.h>
+#include <tvm/runtime/module.h>
+#include <treelite/predictor.h>
+#include <treelite/c_api_runtime.h>
+#include "../3rdparty/treelite/dmlc-core/src/io/filesys.h"
+//#include <graph/graph_runtime.cc>
+//#include <graph/debug/graph_runtime_debug.cc>
+#include <runtime_base.h>
+#include <dlpack/dlpack.h>
+#include <string>
+#include <vector>
+#include <sys/types.h>
+#include "dlr.h"
+
+#define LINE_SIZE 256
+
+
+typedef struct {
+  std::string model_lib;
+  std::string params;
+  std::string model_json;
+  std::string ver_json;
+} ModelPath;
+
+void listdir(const std::string& dirname, std::vector<std::string> &paths);
+
+inline bool endsWith(const std::string &mainStr, const std::string &toMatch)
+{
+	if(mainStr.size() >= toMatch.size() &&
+	  mainStr.compare(mainStr.size() - toMatch.size(), toMatch.size(), toMatch) == 0)
+		return true;
+	else
+		return false;
+}
+
+enum class DLRBackend {
+  kTVM,
+  kTREELITE
+};
+/*! /brief Get the paths of the TVM model files.
+ */
+ModelPath get_tvm_paths(const std::string &tar_path);
+
+/*! /brief Get the paths of the Treelite model files.
+ */
+ModelPath get_treelite_paths(const std::string& dirname);
+
+/*! /brief Get the backend based on the contents of the model folder.
+ */
+DLRBackend get_backend(const std::string &dirname);
```

Comment:
Let's use the left pointer alignment style (`const std::string& dirname`) to keep the consistency.